models/BaseModel.js

```diff
@@ -57,29 +57,31 @@ class BaseModel extends Model {
   }
 
   $afterGet (queryOptions /*: any */, context /*: any */) /*: any */ {
-    const parent = super.$beforeUpdate(queryOptions, context)
+    const parent = super.$afterGet(queryOptions, context)
     let doc = this
     return Promise.resolve(parent).then(function () {
       doc = doc.formatIdsToUrl(doc, doc.getType())
     })
   }
 
   $afterInsert (queryOptions /*: any */, context /*: any */) /*: any */ {
-    const parent = super.$beforeUpdate(queryOptions, context)
+    const parent = super.$afterInsert(queryOptions, context)
     let doc = this
     return Promise.resolve(parent).then(function () {
       doc = doc.formatIdsToUrl(doc, doc.getType())
     })
   }
 
   $beforeInsert (queryOptions /*: any */, context /*: any */) /*: any */ {
-    const parent = super.$beforeUpdate(queryOptions, context)
+    const parent = super.$beforeInsert(queryOptions, context)
     let doc = this
     return Promise.resolve(parent).then(function () {
       doc.id = crypto.randomBytes(16).toString('hex')
       const time = new Date().toISOString()
       doc.published = time
-      doc.updated = time
+      if (doc.constructor.name !== 'ReadActivity') {
+        doc.updated = time
+      }
```

Comment:
actually, I fixed the 'updated' issue in another way in my pull request. You can remove those three lines completely.